File: front-end/src/app/service-models/app/v1/client/generate-payroll-transaction-s-m.ts
import { PayrollTransactionSM } from './payroll-transaction-s-m';
import { EmployeeStatusSM } from '../../enums/employee-status-s-m.enum';

export class GeneratePayrollTransactionSM extends PayrollTransactionSM {
    employeeName!: string;
    designation!: string;
    employeeStatus!: EmployeeStatusSM;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/GeneratePayrollTransactionSM.cs
using Siffrum.Web.Payroll.ServiceModels.Enums;

namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{
    public class GeneratePayrollTransactionSM : PayrollTransactionSM
    {
        public string EmployeeName { get; set; }
        public string Designation { get; set; }
        public EmployeeStatusSM EmployeeStatus { get; set; }
    }
}
